libraries/botframework-connector/src/auth/endorsementsValidator.ts

```diff
@@ -25,14 +25,14 @@ export class EndorsementsValidator {
      * some specific channels. That list is the endorsement list, and is validated here against the channelId.
      * @returns {boolean} True is the channelId is found in the Endorsement set. False if the channelId is not found.
      */
-    static validate(channelId: string, endorsements: string[]): boolean {
+    static validate(channelId: string, endorsements: Set<string>): boolean {
```

Comment:
Changing the function's signature would break the backward compat. Instead, could you create a new function that receives the Set argument and call it inside the old validate function?

Reply:
I'm not opposed to breaking this if having another method is clutter.  Is there harm in having two?

Reply:
I suggest a private function to be called inside this old one (passing the converted strings[] as set<string>) so those who are using this with an array won't experience a break.